src/amqp-client/channel.cr

```diff
@@ -256,24 +280,35 @@ class AMQP::Client
       if @confirm_mode
         @confirm_lock.synchronize do
           msgid = @confirm_id &+= 1_u64
-          @unconfirmed_publishes.push msgid
+          @unconfirmed_publishes.push({msgid, blk})
           msgid
         end
       else
+        raise ArgumentError.new("On confirm block received but channel is not in confirm mode") if blk
```

Comment:
Right now we raise if a block is given to `basic_publish` but the channel isn't in confirm mode, should we instead automatically enable it?